### PR TITLE
Add debit transfer function to withdraw debt from debtor to creditor

### PR DESCRIPTION
**Pull request by cducrest**
Closes https://github.com/trustlines-protocol/contracts/issues/265

### CODE REVIEW

File: contracts/CurrencyNetwork.sol [hunk 1]
```diff
@@ -267,6 +268,48 @@ contract CurrencyNetwork is CurrencyNetworkInterface, Ownable, Authorizable, Des
...
+            _path);
+
+        if (success) {
+            emit Transfer(
+                _from,
+                _to,
+                _value,
+                _extraData
+            );
+            _reduceDebt(_from, _to, _value);
```
Comment (berndbohmeier) [inline contracts/CurrencyNetwork.sol]: We should first do this. We dont call external stuff at the moment, but I would rather still follow that pattern. 
We can just revert on success false.

File: contracts/CurrencyNetwork.sol [hunk 2]
```diff
@@ -267,6 +268,48 @@ contract CurrencyNetwork is CurrencyNetworkInterface, Ownable, Authorizable, Des
...
     }
 
+    /**
+     * @notice send `_value` token to `_to` from `_from`
+     * `_from` needs to have a debt towards è_to` of at least `_value`
+     * `_to` needs to be msg.sender
+     * @param _from The address of the sender
+     * @param _to The address of the recipient
+     * @param _value The amount of token to be transferred
+     * @param _maxFee Maximum fee the sender wants to pay
```
Comment (berndbohmeier) [inline contracts/CurrencyNetwork.sol]: receiver pays the fees says the code?

File: contracts/CurrencyNetwork.sol [hunk 3]
```diff
@@ -267,6 +268,48 @@ contract CurrencyNetwork is CurrencyNetworkInterface, Ownable, Authorizable, Des
         }
     }
 
+    /**
+     * @notice send `_value` token to `_to` from `_from`
+     * `_from` needs to have a debt towards è_to` of at least `_value`
```
Comment (berndbohmeier) [inline contracts/CurrencyNetwork.sol]: e_to?

File: contracts/CurrencyNetwork.sol [hunk 4]
```diff
@@ -1423,4 +1471,13 @@ contract CurrencyNetwork is CurrencyNetworkInterface, Ownable, Authorizable, Des
         return keccak256(abi.encodePacked(_creditor, _debtor));
     }
 
+    function _safeSum(int72 a, int72 b) internal pure returns (int72 sum) {
+        sum = a + b;
+        if (a > 0 && b > 0) {
+            assert(sum > 0);
```
Comment (berndbohmeier) [inline contracts/CurrencyNetwork.sol]: require

File: contracts/CurrencyNetwork.sol [hunk 5]
```diff
@@ -1199,6 +1230,23 @@ contract CurrencyNetwork is CurrencyNetworkInterface, Ownable, Authorizable, Des
         );
     }
 
+    function _increaseDebt(address debtor, address creditor, int72 value) internal {
```
Comment (berndbohmeier) [inline contracts/CurrencyNetwork.sol]: if you can actually reduce debt with it, maybe better call it addDebt

Comment (cducrest) [reply contracts/CurrencyNetwork.sol]: I call it `addToDebt`

**Pull request merged**